presto-main/src/main/java/com/facebook/presto/operator/scalar/SessionFunctions.java

```diff
@@ -33,4 +34,12 @@ public static Slice currentUser(ConnectorSession session)
     {
         return utf8Slice(session.getUser());
     }
+
+    @ScalarFunction(value = "$current_path", hidden = true)
+    @Description("retrieve current path")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice currentPath(ConnectorSession session)
+    {
+        return utf8Slice(requireNonNull(session.getPath(), "session path is null"));
```

Comment:
Can `session.getPath()` ever be `null`? If yes, instead of throwing a NPE (which `requireNonNull` will do) we should do `throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Session path is null");`.